ax/core/utils.py

```diff
@@ -148,21 +150,37 @@ def best_feasible_objective(  # pragma: no cover
     return accumulate(f)
 
 
+def _extract_generator_run(trial: BaseTrial) -> GeneratorRun:
+    if isinstance(trial, BatchTrial):
+        return trial._generator_run_structs[0].generator_run
```

Comment:
This assumes the trial was generated by one generator run (in some cases there may be more, like in the online experimentation setting). We should probably raise an informative `UnsupportedError` or `NotImplementedError` here if that's not the case (and we don't want to implement the general case for now).